src/DIRAC/WorkloadManagementSystem/Utilities/ParametricJob.py

```diff
@@ -51,7 +51,7 @@ def getParameterVectorLength(jobClassAd):
       else:
         nThisParValues = jobClassAd.getAttributeInt(attribute)
       if nParValues is not None and nParValues != nThisParValues:
-        return S_ERROR(EWMSJDL, "Different length of parameter vectors: for %s, %d != %d" % (attribute,
+        return S_ERROR(EWMSJDL, "Different length of parameter vectors: for %s, %s != %d" % (attribute,
```

Comment:
why ? It is supposed to be an `int` too no ?

Reply:
It starts off as `nParValues = None` and pylint isn't smart enough to understand it definitely isn't `None` anymore because of the `if`. I'm pretty sure this change is harmless for the string and still allows it to validate the other parameters in the string to still be validated.

Reply:
but then I would advocate to disable locally pylint rather than changing the actual code to please a static analyzer

Reply:
If it's disabled locally it would miss bugs like:
```suggestion
        return S_ERROR(EWMSJDL, "Different length of parameter vectors: for %d, %s != %d" % (attribute,
```
There are limits to how many changes you should make (https://github.com/DIRACGrid/DIRAC/pull/4990#discussion_r582139236 is an example of too far) but this seems harmless enough.